### ISSUE TITLE
Support Azure SQL Serverless edition

### ISSUE THREAD

**Issue by StefanSchoof** [terraform-providers/terraform-provider-azurerm#4185]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure SQL has a serverless edition. It would be great if this is possible to setup with terraform.

### New or Affected Resource(s)

* azurerm_sql_database

### Potential Terraform Configuration

```hcl
resource "azurerm_sql_database" "test" {
  name = "mysqldatabase"
  resource_group_name = "${azurerm_resource_group.test.name}"
  location = "West US"
  server_name = "${azurerm_sql_server.test.name}"
  edition = "Serverless"
  max_vcore = 3
  min_vcore = 1
  auto_pause_delay_in_minutes = 720
 }
```

### References

* 

> https://docs.microsoft.com/en-us/azure/sql-database/sql-database-serverless

**Comment by tombuildsstuff** [comment]
hi @StefanSchoof 

Thanks for opening this issue :)

Taking a look through this appears to be a duplicate of #1747 - rather than having multiple issues open tracking the same thing I'm going to close this issue in favour of that one; would you mind subscribing to #1747 for updates?

Thanks!

**Comment by StefanSchoof** [comment]
Thanks for the quick reply.
I do not thing this is a duplicate. In the whole #1747 is no mention of serverless. Do you think #1747 should also add the serverless edition (it is in preview)?

**Comment by tombuildsstuff** [comment]
@StefanSchoof thanks for the clarification - I'd agree this makes sense to re-open/track separately

**Comment by craluri** [comment]
Hi Tombuild,

Do you have update on the above mentioned information?

**Comment by kszicsillag** [comment]
SQL Serverless is now out of preview: https://azure.microsoft.com/en-us/updates/azure-sql-database-serverless-is-now-generally-available/

**Comment by RavulaChetan** [comment]
Do we Terraform support to create database ComputeModel Serverless yet?

**Comment by DaRosenberg** [comment]
29 upvotes for this request, and this Azure service has now been in GA for 3 months. Any chance the folks from Terraform could provide some advice on where this currently is in the planning?

**Comment by abhidhar82** [comment]
I was able to create a serverless SQL DB using edition as "GeneralPurpose" and requested_service_objective_name with value like "GP_S_Gen5_2" i.e. Gen5 and vcpu =2.

Sample:
resource "azurerm_sql_database" "testdb" {
  name                = "testdb"
  resource_group_name = "testrg"
  location            =  ####
  server_name         = #####
  create_mode         = "Default"
  edition             = "GeneralPurpose"
  max_size_bytes      = "34359738368"
  requested_service_objective_name = "GP_S_Gen5_${var.dbvcpu}"
}
Hope this helps.

**Comment by DaRosenberg** [comment]
@abhidhar82 that is definitely helpful! One question though - the vCore number you use, does that get used as the minimum or the maximum? With SQL Serverless, a range between min and max number of vCores can be specified, so just wondering which would be referred to here, and how to set the other one?

**Comment by abhidhar82** [comment]
@DaRosenberg good question. I have observed if you set the vCore to 4 then the min is set 2 and max 4 with autopause as 1 hour. I wasn't able to find away to set these.

**Comment by tombuildsstuff** [comment]
Fixed via #6083
